### PR TITLE
Restore the intercom button!

### PR DESCRIPTION
**Pull request by discorick**

### CODE REVIEW

File: app/views/shared/_analytics.html.erb [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
   window.analytics.load("<%= ENV["SEGMENTIO_KEY"] %>");
-  window.analytics.page({ userId: '<% current_user.id %>' });
- <% end %>
+  <% if logged_in? %>
+    window.analytics.identify('<%= current_user.id %>', <%= current_user.attribs.to_json.html_safe %>)
+    window.analytics.page({ userId: '<% current_user.id %>' });
+  <% else %>
+    window.analytics.page();
+  <% end %>
```
Comment (rauhryan) [inline app/views/shared/_analytics.html.erb]: We need to be very careful with the identify call

if we send null as the email field, intercom will pave over that attribute

perhaps we shouldn't send any traits at all on the client side?

File: app/views/shared/_analytics.html.erb [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
   window.analytics.load("<%= ENV["SEGMENTIO_KEY"] %>");
-  window.analytics.page({ userId: '<% current_user.id %>' });
- <% end %>
+  <% if logged_in? %>
+    window.analytics.identify('<%= current_user.id %>', <%= current_user.attribs.to_json.html_safe %>)
+    window.analytics.page({ userId: '<% current_user.id %>' });
```
Comment (dahlbyk) [inline app/views/shared/_analytics.html.erb]: This should be redundant with `identify` restored. Maybe just reset to exactly what we had before?

**Comment by discorick** [comment]
Ok, it might just be in our best interest to start using https://docs.intercom.io/install-on-your-web-product/use-javascript-to-install-intercom-on-any-web-site on the client side.

I am gonna look into it further.

**Comment by rauhryan** [comment]
> Ok, it might just be in our best interest to start using https://docs.intercom.io/install-on-your-web-product/use-javascript-to-install-intercom-on-any-web-site on the client side.
> 
> I am gonna look into it further.

I don't want this to hold up simply getting the button to reappear, can we start a new thread to discuss getting Acquire installed

**Comment by discorick** [comment]
Alrighty! After some local testing it appears passing in a blank object in client-side `identify` calls does not trigger any kind of property overwrite with the intercom integration :+1:

**Comment by dahlbyk** [comment]
:shipit:

**Pull request merged**